src/Admin/AbstractAdmin.php

```diff
@@ -2132,7 +2133,7 @@ public function getTranslationDomain()
      *
      * @deprecated since sonata-project/admin-bundle 3.9, to be removed with 4.0
      */
-    public function setTranslator(TranslatorInterface $translator)
+    public function setTranslator(LegacyTranslatorInterface $translator)
```

Comment:
Should this method become deprecated in favor of `setContractTranslator`?

Reply:
I'd prefer not polluting our API for reasons like this. IMO, the better way would be tell the user that in the next major, `setTranslator()` will change the required type for their argument 1.

Reply:
I must back over my own words. `AbstractAdmin::setTranslator()` is deprecated too.